packages/esm-appointments-app/src/hooks/useAppointmentsCalendar.ts

```diff
@@ -66,3 +66,124 @@ function evaluateAppointmentCalendarDates(forDate: string, period: string) {
     };
   }
 }
+export async function refreshAppointmentCalendar(forDate: string, period: string, serviceName: string) {
+  const { startDate, endDate } = evaluateAppointmentCalendarDates(forDate, period);
+  const url = `${restBaseUrl}/appointment/appointmentSummary?startDate=${startDate}&endDate=${endDate}`;
+  try {
+    const response = await openmrsFetch(url);
+    const data = response?.data || [];
+    for (const service of data) {
+      if (service.appointmentService?.name === serviceName) {
+        for (const [date, value] of Object.entries(service.appointmentCountMap || {})) {
+          if (date === forDate) {
+            const count =
+              value && typeof value === 'object' && 'allAppointmentsCount' in value
+                ? value.allAppointmentsCount
+                : undefined;
+            if (typeof count === 'number') {
+              return;
+            }
+          }
+        }
+      }
+    }
+  } catch (error) {
+    return;
+  }
+}
+
+export async function updateAppointmentService(
+  uuid: string,
+  payload: any,
+): Promise<{
+  success: boolean;
+  data?: any;
+  error?: { status: number; message: string; code: string; details?: any };
+}> {
+  try {
+    const existingService = await fetchExistingService(uuid);
+    const mergedPayload = buildUpdatePayload(uuid, payload, existingService);
+    const cleanPayload = removeEmptyValues(mergedPayload);
+    const response = await openmrsFetch(`${restBaseUrl}/appointmentService`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(cleanPayload),
+    });
+
+    return {
+      success: true,
+      data: response.data,
+    };
+  } catch (err) {
+    return {
+      success: false,
+      error: {
+        status: err?.response?.status || 500,
+        message: err?.message || 'Failed to update appointment service',
+        code: 'UPDATE_FAILED',
+        details: err?.responseBody || err?.response?.data,
+      },
+    };
+  }
+}
+
+async function fetchExistingService(uuid: string): Promise<any> {
+  try {
+    const response = await openmrsFetch(`${restBaseUrl}/appointmentService?uuid=${uuid}`);
+    return response.data || {};
+  } catch (err) {
+    return {};
+  }
+}
+
+function buildUpdatePayload(uuid: string, payload: any, existingService: any): any {
```

Comment:
thanks @gabriel090 for the work on this , but the function would benefit from stronger typing because right now both payload and `existingService` are typed as any, which removes typeScript’s safety guarantees and makes the function harder to reason about.

Defining interfaces (e.g., `ServicePayload`, `ExistingService`) would help:
prevent silent bugs from incorrect or missing fields ,give better IntelliSense and documentation for future contributors, make refactoring safer when backend models evolve, clarify which fields are optional vs required

Adding these interfaces will make the function more predictable and easier to maintain in the long term.